modules/foreman_search_facts.py

```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# (c) 2018, Sean O'Keeffe <seanokeeffe797@gmail.com>
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+ANSIBLE_METADATA = {'metadata_version': '1.0',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: foreman_search_facts
+short_description: Gather facts about Foreman Entities
+description:
+  - "Gather facts about Foreman Entites"
```

Comment:
'Entities' is verbiage from Nailgun. In Foreman I think we'd just call them foreman resources.

Reply:
Most of the time i think about 'unprocessible entity'. ;)
Ressource sounds good to me.

Reply:
I'm fine with 'entity' if both of you prefer it. :)

Reply:
No - think resource is clearer :+1: